src/pymoca/ast.py

```diff
@@ -908,45 +926,111 @@ def __deepcopy__(self, memo):
         return new
 
     def __repr__(self):
-        return "{}(type={!r}, name={!r})".format(type(self).__name__, self.type, self.name)
+        return "{}(name={!r}, type={!r})".format(type(self).__name__, self.name, self.type)
 
     def __str__(self):
         return '{} {}, Type "{}"'.format(type(self).__name__, self.name, self.type)
 
 
-class InstanceClass(Class):
+class InstanceElement:
+    """
+    Base class for instance elements (symbols, classes, extends).
```

Comment:
I'm a bit confused (at this stage of the review) why/how "extends clauses" can be instantiated.

Reply:
Short answer to "why" is because they are not resolved until the flattening stage according to the spec. The original issue that prompted this PR was losing the original scope of the extends. There is also the issue of maintaining order of elements in the class definition that the spec mentions that the current implementation can't do without a more significant tear-up of the AST and parser.

Reply:
Hopefully things get a little more clear as you work your way through tree.py. 🤞